File: src/test/java/de/espend/idea/php/annotation/tests/AnnotationUsageIndexTest.java
package de.espend.idea.php.annotation.tests;

import de.espend.idea.php.annotation.AnnotationUsageIndex;

import java.io.File;

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 * @see de.espend.idea.php.annotation.AnnotationUsageIndex
 */
public class AnnotationUsageIndexTest extends AnnotationLightCodeInsightFixtureTestCase {
    public void setUp() throws Exception {
        super.setUp();
        myFixture.copyFileToProject("classes.php");
        myFixture.copyFileToProject("usages.php");
    }

    public String getTestDataPath() {
        return "src/test/java/de/espend/idea/php/annotation/tests/fixtures";
    }

    public void testThatUsagesAreInIndex() {
        assertIndexContains(AnnotationUsageIndex.KEY, "Doctrine\\ORM\\Mapping\\Embedded");
    }
}

File: src/test/kotlin/de/espend/idea/php/annotation/tests/AnnotationUsageIndexTest.kt
package de.espend.idea.php.annotation.tests

import de.espend.idea.php.annotation.AnnotationUsageIndex

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 * @see de.espend.idea.php.annotation.AnnotationUsageIndex
 */
class AnnotationUsageIndexTest : AnnotationLightCodeInsightFixtureTestCase() {
    public override fun setUp() {
        super.setUp()
        myFixture.copyFileToProject("classes.php")
        myFixture.copyFileToProject("usages.php")
    }

    override fun getTestDataPath(): String {
        return "src/test/java/de/espend/idea/php/annotation/tests/fixtures"
    }

    fun testThatUsagesAreInIndex() {
        assertIndexContains(AnnotationUsageIndex.KEY, "Doctrine\\ORM\\Mapping\\Embedded")
        assertIndexContains(AnnotationUsageIndex.KEY, "My\\Route")
    }
}
